google/cloud/sql/connector/connector.py

```diff
@@ -238,6 +239,12 @@ async def connect_async(
         # attempt to make connection to Cloud SQL instance
         try:
             instance_data, ip_address = await instance.connect_info(ip_type)
+            # resolve DNS name into IP address for PSC
+            if ip_type.value == "PSC":
+                addr_info = await self._loop.getaddrinfo(
+                    ip_address, None, family=socket.AF_INET
+                )
+                ip_address = addr_info[0][4][0]
```

Comment:
A comment here about these numbers would help. Also, do you see any reason to guard against these indexes being null?

Reply:
I'll throw in an error message if the list is empty saying "Could not resolve PSC DNS name into IP addresss" or something